internal/runbits/buildscript/file.go

```diff
@@ -47,33 +50,34 @@ type primeable interface {
 	primer.SvcModeler
 }
 
-func Initialize(path string, auth *authentication.Auth, svcm *model.SvcModel) error {
-	scriptPath := filepath.Join(path, constants.BuildScriptFileName)
-	script, err := ScriptFromFile(scriptPath)
-	if err == nil {
-		return nil // nothing to do, buildscript already exists
-	}
-	if !errors.Is(err, os.ErrNotExist) {
-		return errs.Wrap(err, "Could not read build script from file")
-	}
-
-	logging.Debug("Build script does not exist. Creating one.")
-	commitId, err := localcommit.Get(path)
+// Initialize creates a new build script for the local project. It will overwrite an existing one so
+// commands like `state reset` will work.
+func Initialize(proj projecter, auth *authentication.Auth, svcm *model.SvcModel) error {
+	logging.Debug("Initializing build script")
+	commitId, err := localcommit.Get(proj.Dir())
 	if err != nil {
 		return errs.Wrap(err, "Unable to get the local commit ID")
 	}
 
 	buildplanner := buildplanner.NewBuildPlannerModel(auth, svcm)
-	script, err = buildplanner.GetBuildScript(commitId.String())
+	script, err := buildplanner.GetBuildScript(commitId.String())
 	if err != nil {
 		return errs.Wrap(err, "Unable to get the remote build expression and time")
 	}
 
+	if url, err := projectURL(proj, commitId.String()); err == nil {
+		script.SetProject(url)
+	} else {
+		return errs.Wrap(err, "Unable to set project")
+	}
```

Comment:
No, I cannot because the projecter has the old commit ID from before a mutation. localcommit.Set() updates the URL on disk, but the primed project was not reloaded/reset/whatever, so its URL remains the same as it was originally.

I don't really see the point in changing this because when we replace project file with build scripts, we'll have to do it again. I thought the point of this runbit was to know as little about projects as possible.

Reply:
Good point. I don't like how opinionated the current solution is wrt the formatting of the URL though. I guess what this is touching on is that we probably need some central mechanic for project URLs. Still, to avoid further churn I think for this story we can maybe have a middle ground; like use `project.URL()` and only splice in the commitID. So you'd avoid having any opinion on the URL beyond the commitID parameter.